nisystemlink/clients/assetmanagement/models/_asset.py

```diff
@@ -0,0 +1,112 @@
+from typing import Dict, List, Optional
+
+from nisystemlink.clients.core._uplink._json_model import JsonModel
+
+from ._asset_calibration import (
+    CalibrationStatus,
+    ExternalCalibration,
+    SelfCalibration,
+    TemperatureSensor,
+)
+from ._asset_location import AssetLocation
+from ._asset_types import AssetBusType, AssetDiscoveryType, AssetType
+
+
+class Asset(JsonModel):
+    """Model for an object describing an asset with all of its properties."""
+
+    model_name: Optional[str] = None
+    """Gets or sets model name of the asset."""
+
+    model_number: int
+    """Gets or sets model number of the asset."""
+
+    serial_number: Optional[str] = None
+    """Gets or sets serial number of the asset."""
+
+    vendor_name: Optional[str] = None
+    """Gets or sets vendor name of the asset."""
+
+    vendor_number: int
+    """Gets or sets vendor number of the asset."""
+
+    bus_type: AssetBusType
+    """Gets or sets all supported bus types for an asset."""
+
+    name: Optional[str] = None
+    """Gets or sets name of the asset."""
+
+    asset_type: AssetType
+    """Gets or sets all supported asset types."""
+
+    discovery_type: AssetDiscoveryType
+    """Gets or sets the discovery type."""
+
+    firmware_version: Optional[str] = None
+    """Gets or sets firmware version of the asset."""
+
+    hardware_version: Optional[str] = None
+    """Gets or sets hardware version of the asset."""
+
+    visa_resource_name: Optional[str] = None
+    """Gets or sets VISA resource name of the asset."""
+
+    temperature_sensors: Optional[List[TemperatureSensor]] = None
+    """Gets or sets an array of temperature sensor information."""
+
+    supports_self_calibration: bool
+    """Gets or sets whether the asset supports self-calibration."""
+
+    supports_external_calibration: bool
+    """Gets or sets whether the asset supports external calibration."""
+
+    custom_calibration_interval: Optional[int] = None
+    """Gets or sets the interval represented in months used for computing calibration due date."""
+
+    self_calibration: Optional[SelfCalibration] = None
+    """Gets or sets the last self-calibration of the asset."""
+
+    is_n_i_asset: bool
```

Comment:
```suggestion
    is_NI_asset: bool
```

You might need to force a specific alias to get Pydantic to parse this properly (plus projection support)

```suggestion
    is_NI_asset: Optional[bool] = Field(alias="isNIAsset", default=None)
```

Reply:
Updated as suggested